### PR TITLE
robot_name

### PR DESCRIPTION
**Pull request by zsyed91**
Refactor robot_name to use methods and clean up the code

### CODE REVIEW

File: robot_name/robot_name.rb [hunk 1]
```diff
@@ -2,24 +2,51 @@ class NameCollisionError < RuntimeError; end
...
+    @@registry << name
+  end
 
-      @name = "#{generate_char.call}#{generate_char.call}#{generate_num.call}#{generate_num.call}#{generate_num.call}"
+  def generate_name
+    if name_generator
+      name_generator.call
+    else
+      generate_character_sequence(2) + generate_number_sequence(3)
     end
```
Comment (djkotowski) [inline robot_name/robot_name.rb]: Prefer guard clauses to decision blocks here.  Also, prefer `<<` to `+` (the former appends to the first string, the latter combines into a new string)
```ruby
return name_generator.call if name_generator
generate_character_sequence(2) << generate_number_sequence(3)
```

**Comment by djkotowski** [comment]
Nice work!  You're done with this one.

File: robot_name/robot_name.rb [hunk 2]
```diff
@@ -2,24 +2,51 @@ class NameCollisionError < RuntimeError; end
...
+  def generate_character
+    ('A'..'Z').to_a.sample
+  end
+
+  def generate_number
+    rand(10).to_s
+  end
+
+  def check_name_collision
+    return if name =~ /[[:alpha]]{2}[[:digit:]]{3}/
```
Comment (kotp) [inline robot_name/robot_name.rb]: Does line 49 ever get reached if this is valid?  Does line 47?

Comment (djkotowski) [inline robot_name/robot_name.rb]: Yep, the logic is a bit messed up here.

File: robot_name/robot_name.rb [hunk 3]
```diff
@@ -2,24 +2,51 @@ class NameCollisionError < RuntimeError; end
 
 class Robot
   attr_accessor :name
+  attr_accessor :name_generator
```
Comment (djkotowski) [inline robot_name/robot_name.rb]: Just combine the two accessors into one line.

**Pull request closed without merging**